packages/backend/src/open_payments/payment_pointer/service.test.ts

```diff
@@ -133,6 +134,86 @@ describe('Open Payments Payment Pointer Service', (): void => {
     })
   })
 
+  describe('Update Payment Pointer', (): void => {
+    test.each`
+      initialIsActive | status        | expectedIsActive
+      ${true}         | ${undefined}  | ${true}
+      ${true}         | ${'INACTIVE'} | ${false}
+      ${false}        | ${'ACTIVE'}   | ${true}
+      ${false}        | ${undefined}  | ${false}
+    `(
+      'Payment pointer  with initial isActive of $initialIsActive can be updated with status of $status and publicName of $publicName',
+      async ({ initialIsActive, status, expectedIsActive }): Promise<void> => {
+        const paymentPointer = await createPaymentPointer(deps)
+
+        if (!initialIsActive) {
+          await paymentPointer.$query(knex).patch({ deactivatedAt: new Date() })
+        }
+
+        const updatedPaymentPointer = await paymentPointerService.update({
+          id: paymentPointer.id,
+          status
+        })
+        assert.ok(!isPaymentPointerError(updatedPaymentPointer))
+
+        expect(updatedPaymentPointer.isActive).toEqual(expectedIsActive)
+
+        await expect(
+          paymentPointerService.get(paymentPointer.id)
+        ).resolves.toEqual(updatedPaymentPointer)
+      }
+    )
+
+    describe('Payment pointer fields can be updated independently', (): void => {
+      let paymentPointer: PaymentPointer
+      const initialName = 'Initial Name'
+
+      beforeEach(async (): Promise<void> => {
+        paymentPointer = await createPaymentPointer(deps, {
+          publicName: initialName
+        })
+      })
+
+      test('publicName', async (): Promise<void> => {
+        const newName = 'New Name'
+        const updatedPaymentPointer = await paymentPointerService.update({
+          id: paymentPointer.id,
+          publicName: newName
+        })
+        assert.ok(!isPaymentPointerError(updatedPaymentPointer))
+        expect(updatedPaymentPointer.deactivatedAt).toEqual(null)
+        expect(updatedPaymentPointer.publicName).toEqual(newName)
+        await expect(
+          paymentPointerService.get(paymentPointer.id)
+        ).resolves.toEqual(updatedPaymentPointer)
+      })
+
+      test('status', async (): Promise<void> => {
+        const updatedPaymentPointer = await paymentPointerService.update({
+          id: paymentPointer.id,
+          status: PaymentPointerStatus.Inactive
+        })
+        assert.ok(!isPaymentPointerError(updatedPaymentPointer))
+        expect(updatedPaymentPointer.deactivatedAt).toBeDefined()
+        expect(updatedPaymentPointer.isActive).toEqual(false)
+        expect(updatedPaymentPointer.publicName).toEqual(initialName)
+        await expect(
+          paymentPointerService.get(paymentPointer.id)
+        ).resolves.toEqual(updatedPaymentPointer)
+      })
+    })
```

Comment:
I think we're already testing this with the test above IMO. We can just remove the `describe('Payment pointer fields can be updated independently'` block, to have a simple `test('can update publicName', ...)` maybe?

Reply:
yeah you're right, it's pure overlap now. I refactored as you described.